test/functional/specs/C14409.js

```diff
@@ -0,0 +1,65 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import fixtureFactory from "../helpers/fixtureFactory";
+import cookies from "../helpers/cookies";
+import alloyEvent from "../helpers/alloyEvent";
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title: "C14409 - Consenting to no purposes should be persisted.",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14409",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const setConsentToOut = ClientFunction(() => {
+  return window.alloy("setConsent", { general: "out" });
+});
+
+test("C14409 - Consenting to no purposes should be persisted.", async () => {
+  const imsOrgId = "53A16ACB5CC1D3760A495C99@AdobeOrg";
+  await cookies.clear();
+
+  // configure alloy
+  const configure = await alloyEvent("configure", {
+    configId: "9999999",
+    orgId: imsOrgId,
+    debugEnabled: true,
+    idMigrationEnabled: false
+  });
+
+  await configure.promise;
+
+  // set consent to out
+  await setConsentToOut();
+
+  // Reload page and reconfigure alloy
+  // [TODO] Navigate to a different subdomain when it is available
+  // https://github.com/DevExpress/testcafe/blob/a4f6a4ac3627ebeb29b344ed3a1793627dd87909/docs/articles/documentation/test-api/actions/navigate.md
+  await t.eval(() => document.location.reload());
+
+  const reconfigure = await alloyEvent("configure", {
+    configId: "9999999",
+    orgId: imsOrgId,
+    debugEnabled: true,
+    idMigrationEnabled: false
+  });
```

Comment:
I recommend taking this moving this into a function above the test. Something like this:

```
const configureAlloy = async () => {
  const configure = await alloyEvent("configure", {
    configId: "9999999",
    orgId: imsOrgId,
    debugEnabled: true,
    idMigrationEnabled: false
  });
  await configure.promise;
});
```

Then inside the test in the two places where you configure Alloy, just do `await configureAlloy()`;

Reply:
Actually, I don't think you need the `await configure.promise` either. Just this:

```
const configureAlloy = () => {
  return alloyEvent("configure", {
    configId: "9999999",
    orgId: imsOrgId,
    debugEnabled: true,
    idMigrationEnabled: false
  });
});
```